File: app/src/main/java/com/divercity/app/db/DateTypeConverter.java
package com.divercity.app.db;

import android.arch.persistence.room.TypeConverter;

import com.divercity.app.core.utils.Util;

/**
 * Created by lucas on 02/01/2019.
 */

public class DateTypeConverter {

    @TypeConverter
    public static Long toLong(String value) {
        return Util.getMilisecFromStringDate(value);
    }

    @TypeConverter
    public static String toString(Long value) {
        return "pol";
    }
}

File: app/src/main/java/com/divercity/android/db/converter/DateTypeConverter.kt
package com.divercity.android.db.converter

import androidx.room.TypeConverter
import java.util.*

/**
 * Created by lucas on 16/01/2019.
 */

object DateTypeConverter {

    @TypeConverter
    @JvmStatic
    fun toLong(value: Date?): Long {
        return value?.time ?: Date().time
    }

    @TypeConverter
    @JvmStatic
    fun toString(value: Long): Date {
        return Date(value)
    }
}